### PR TITLE
[Merged by Bors] - handle peer state transitions on gossipsub score changes + refactoring

### PR DESCRIPTION
**Pull request by blacktemplar**
## Issue Addressed

NA

## Proposed Changes

Correctly handles peer state transitions on gossipsub changes + refactors handling of peer state transitions into one function used for lighthouse score changes and gossipsub score changes.

### CODE REVIEW

File: beacon_node/eth2_libp2p/src/peer_manager/mod.rs [hunk 1]
```diff
@@ -163,55 +163,30 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
...
+                Self::handle_score_transitions(
+                    previous_state,
+                    peer_id,
+                    info,
+                    &mut to_ban_peers,
+                    &mut to_unban_peers,
+                    &mut self.events,
+                    &self.log,
+                );
+                if previous_state == info.score_state() {
```
Comment (AgeManning) [inline beacon_node/eth2_libp2p/src/peer_manager/mod.rs:183]: Should this be a `!=` ?

**Review by AgeManning — CHANGES_REQUESTED** [review]

**Review by AgeManning — APPROVED** [review]
Updated

File: beacon_node/eth2_libp2p/src/peer_manager/mod.rs (hunk 1 above)
Comment (blacktemplar) [reply beacon_node/eth2_libp2p/src/peer_manager/mod.rs]: No this should really be an `==`. It was the else branch in the original code. Basically we log already a line inside `handle_score_transitions` if the state changed. This line here is to log that the score changed if the state didn't change.

**Pull request closed without merging**